File: backend/app/schemas/summarize.py
"""Schemas for the /channels/{channel_id}/summarize endpoint."""

from __future__ import annotations

from pydantic import BaseModel


class SummarizeResponse(BaseModel):
    """AI-generated thread summary."""

    bullets: list[str]
    decisions: list[str]
    action_items: list[str]
    message_count: int
    cached: bool

File: src/api/summarize.ts
import { apiClient } from './client'

export interface SummarizeResult {
  bullets: string[]
  decisions: string[]
  action_items: string[]
  message_count: number
  cached: boolean
}

export const summarizeApi = {
  async summarize(channelId: string): Promise<SummarizeResult> {
    const { data } = await apiClient.post<{ data: SummarizeResult }>(
      `/channels/${encodeURIComponent(channelId)}/summarize`,
    )
    return data.data
  },
}
